test/test_ops.py

```diff
@@ -1,374 +1,415 @@
-import collections
+from dataclasses import dataclass, field
 import numbers
-from typing import Callable
+from typing import Callable, Dict, List, NamedTuple, Optional, Union
 from torch_xla.core import xla_model as xm
+from torch_xla.core import xla_env_vars as xenv
+from torch_xla.debug import metrics as xmetrics
+import os
 
 import torch
 import unittest
 from torch.testing._internal.common_utils import \
     (TestCase, run_tests)
 from torch.testing._internal.common_methods_invocations import \
-    (SampleInput, op_db)
+    (SampleInput, OpInfo, op_db)
 from torch.testing._internal.common_device_type import \
     (instantiate_device_type_tests, ops)
 
 
-class AllowedOpInfoEntry(
-    collections.namedtuple("AllowedOpInfoEntry",
-                           ["name", "variant_test_name"])):
+@dataclass
+class AllowedOpInfoEntry:
+  name: str
+  variant_test_name: str = ""
 
-  def __new__(cls, name, variant_test_name=""):
-    return super(AllowedOpInfoEntry, cls).__new__(cls, name, variant_test_name)
+
+@dataclass
+class AllowedFallbackOpInfoEntry(AllowedOpInfoEntry):
+  fallback_ops: List[str] = field(default_factory=list)
+  allow_sample: Optional[Callable[[SampleInput], bool]] = None
```

Comment:
What does `allow_sample` mean?

Reply:
It filters the sample list, looking for a specific one. I will leave a comment there.

Reply:
SampleInput is the sample list that you are referring to?

Reply:
`SampleInput` is the class that represents one sample.